src/raftstore/store/worker/split_check.rs

```diff
@@ -204,7 +208,27 @@ impl<C: Sender<Msg>> Runner<C> {
         Some(region_size)
     }
 
-    fn check_split(&mut self, region: &Region) {
+    fn filter_region(&self, region: &Region) -> bool {
+        if self.priority_checker.is_some() {
```

Comment:
if we use table split here, we will regress to old scanning and check every time, this is a bad design.

Reply:
I don't get it. Base on your comment, isn't that what you mean?

> Please see https://github.com/pingcap/tikv/blob/390d99290bee38c7d83e8b5485edd8a2ad5decdb/src/raftstore/store/worker/split_check.rs#L294
> Here we will still check size at size at first.

Reply:
if we must check table split, we should first use a simple algorithm to check whether we need to split the region, not scanning the region. So for a region, you can:

1. Decode the start and end key to see whether they have same table ID.
2. If end key has another table ID, check the end key only has table ID prefix (not a data key)
    2.1 If the end key is a data key, we should check the region
    2.2 If the end key is not, seek to the previous key to see whether the real last key belongs to another table. 

Btw, if we find that the region has only one table, we should use check size later.